flake.nix

```diff
@@ -1,69 +1,76 @@
 # Copyright 2025 The zb Authors
 # SPDX-License-Identifier: MIT
-
 {
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
-    flake-utils.url = "flake-utils";
   };
 
   outputs =
     {
       self,
       nixpkgs,
-      flake-utils,
       ...
     }:
-    flake-utils.lib.eachDefaultSystem (
-      system:
-      let
-        pkgs = import nixpkgs {
-          inherit system;
-        };
+    let
+      inherit (nixpkgs) lib;
 
-        inherit (pkgs.lib.attrsets) optionalAttrs;
+      forEachSystem =
+        fn: lib.genAttrs lib.systems.flakeExposed (system: fn system nixpkgs.legacyPackages.${system});
```

Comment:
[As per nixpkgs](https://github.com/NixOS/nixpkgs/blob/d66143a3c44559a7f90648cdb3efaa54eb0c9784/lib/systems/default.nix#L50-L58):

> This attribute is considered experimental and is subject to change.

I also don't really want to expand to that potential set to what nixpkgs supports: zb is very OS-dependent, and so the exact set of targets is important. Let's use an explicit list of systems.